### ISSUE TITLE
Consider changing marshal default to not length prefix

### ISSUE THREAD

**Issue by ebuchman** [tendermint/go-amino#219]
Currently we have eg. `MarshalBinary` and `MarshalBinaryBare` where the former includes a length prefix and the later does not. It seems more natural to switch the default behaviour to not be length prefixed (afaik proto3 does not auto-include a length prefix), and to have MarshalBinaryLengthPrefix instead of MarshalBinaryBare

**Comment by liamsi** [comment]
I agree, this should be the default behaviour. And you are right, that proto3 doesn't encode the length by default. Length delimiting is used for [streaming] AFAIK. I think some implementations do not even provide length delimited encoding / decoding. Most do though ([java], [prost/rust]).

I think we should change this pre-launch as this would be breaking. Do you agree?

[streaming]: https://developers.google.com/protocol-buffers/docs/techniques#streaming
[java]: https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/Parser#parseDelimitedFrom-java.io.InputStream-
[prost/rust]: https://docs.rs/prost/0.3.2/prost/trait.Message.html#method.encode_length_delimited

**Comment by ebuchman** [comment]
Yes I do! Should get confirmation from @jaekwon too though

**Comment by jaekwon** [comment]
yes :)

**Comment by liamsi** [comment]
First step (rename MarshalBinary) was addressed in #222. Will update depending projects to use the new API. After that we could either rename `MarshalBinaryBare` to `MarshalBinary`, or, we could keep the `Bare` postfix.
